src/reynir_correct/main.py

```diff
@@ -335,14 +337,41 @@ def sentence_stream() -> Iterator[List[CorrectToken]]:
             )
             for ann in a
         ]
+        if args.text:
+            if sent.tree is None:
+                # No need to do more, no grammar errors have been checked
+                print(cleaned, file=args.outfile)
+            else:
+                # We know we have a sent tree, can use that
+                # Use toklist and a
+                cleantoklist: List[CorrectToken] = toklist
+                for xann in a:
+                    if not xann.suggest:
+                        # Nothing to correct with, nothing we can do
+                        continue
+                    if xann.start == xann.end:
+                        # Single-token grammar error, easy to correct
+                        if xann.suggest == "DELETE":
+                            # Token should be deleted.
+                            cleantoklist[xann.start+1].txt = ""
+                        else:
+                            cleantoklist[xann.start+1].txt = xann.suggest
+                    else:
+                        # "Okkur börnunum langar í fisk"
+                        # Only case is one ann, many toks in toklist
+                        allsuggs = xann.suggest.split(" ")
+                        i = 2
+                        for entry in allsuggs:
+                            cleantoklist[xann.start+i].txt = entry
```

Comment:
There is no guarantee that the suggestion has the same number of tokens as the original text... I think this needs to be made more intelligent, i.e. removing a slice from the list and inserting another one instead.

Reply:
Agree. Reversed annotation list to facilitate deleting from the list and deleted other tokens than the first from the list.